### DISCUSSION TITLE
Using Firebase

### DISCUSSION THREAD

**Discussion by dadixon** [pointfreeco/swift-composable-architecture#494] [Q&A]
I'm new to using this architecture so for a newbie discussion, could someone guide me in the right direction to using Firebase Authentication, Storage, and Firestore with TCA? 

I'm also trying to use a Combine version of Firebase: [here](https://github.com/rever-ai/CombineFirebase/tree/master)

My understanding is that you have to populate to Store with data in the Environment before showing the view. If that's the case, I would have to run a query and then set the data in the Environment before showing the view. I'm I thinking about that correctly?

I haven't gotten too deep into the authentication part but I'm assuming after a user logs in, it will store the user in the AppState.

**Comment by andreyz** [comment]
Great place to orient yourself on TCA approach to handling typical use cases such a logging in, fetching data, etc is the Case Studies in the TCA workspace. The Tic-Tac-Toe mini app, has authentication and probably other examples you're looking for. Start there.

As to how to use Firebase, have a look at how ComposableCoreLocation or VoiceRecognition clients are implemented.

**Comment by kdeda** [comment]
State is a snapshot of your data, for example: 
var loginStatus: LoginStatus = LoginStatus.loggedIn or LoginStatus.loggedOut

Reducer is how you modify that state, for example as you handle the login action you can start a url end point request, that when completed will map to an action of loginSuccess.

Action is how you get to the Reducer, so if you have an action loginSuccess, this will be handled by the Reducer and obviously change the loginStatus to LoginStatus.loggedIn

Store is how you send actions from the UI.
